src/actions/targetActions.js

```diff
@@ -10,16 +10,22 @@ const getTargetsSuccess = ({ targets }) => ({
   targets
 });
 
+const deleteTargetSuccess = index => ({
+  type: types.DELETE_TARGET_SUCCESS,
+  index
+});
+
 export const getTargets = () => (dispatch) => {
   targetApi.getTargets()
     .then(targets => dispatch(getTargetsSuccess(targets)))
     .catch(() => toast.error('Could not retrieve targets'));
 };
 
-export const createTarget = target => () => {
+export const createTarget = target => (dispatch) => {
   history.push(routes.index);
   targetApi.createTarget(target)
     .then(() => {
+      dispatch(getTargets());
```

Comment:
Normally, apart from this, you would append the new target to your targets list, so the response in the front-end is immediate.

So, you can have and dispatch a `createTargetSuccess` action  with the newly created target as payload, and use that to append it to the targets list you have in the reducer. I know the response times are quite quick here, but what about a slow network?

Reply:
I thought about that, but what if then the user wants to delete the target? A target ID is needed, which I don't have.

Reply:
Good point 👌